Lib/ufoLib/__init__.py

```diff
@@ -126,8 +97,7 @@ class UFOReader(object):
 	"""
 
 	def __init__(self, path, validate=True):
-		if not os.path.exists(path):
-			raise UFOLibError("The specified UFO doesn't exist.")
+		self.fileSystem = FileSystem(path)
```

Comment:
I think the `path` argument should be changed to `pathOrFileSystem`; if it's a basestring, use it to make `self.fileSystem = FileSystem(path)`, else assume it is a compatible `FileSystem` instance (may be a user-defined subclass that support some alternative, non official, output format like `tar`).

Reply:
hm, perhaps that is already covered, because the `ufoLib.filesystem.FileSystem` class takes either a path string, or another instance of `self.__class__` (or subclasses).

Reply:
no actually, that's not good. that would work like "upcasting" the subclass into an instance of the base FileSystem class. We want to pass an already initialized instance of a FileSystem class or subclass.